drivers/drivers.go

```diff
@@ -125,7 +130,7 @@ const (
 type OSSession interface {
 	OS() OSDriver
 
-	SaveData(ctx context.Context, name string, data io.Reader, meta map[string]string, timeout time.Duration) (string, error)
+	SaveData(ctx context.Context, name string, data io.Reader, fields FileProperties, timeout time.Duration) (string, error)
```

Comment:
Keep in mind that all libs using this need to be updated. Could you make sure to do this on the main ones at least `go-livepeer` and `task-runner` so we don't end up with a breaking update in the future?

Reply:
An argument for getting a pointer here would be just a cleaner migration for everyone already using the lib. Instead of having to change to `FileProperties{}` most of them can just leave as `nil` so not a breaking change on the usage case with default args.

Reply:
Yeah I was considering making it a pointer like you say, will go ahead and do that 👍